waspc/src/Wasp/Generator/FileDraft/CopyDirFileDraft.hs

```diff
@@ -0,0 +1,36 @@
+module Wasp.Generator.FileDraft.CopyDirFileDraft
+  ( CopyDirFileDraft (..),
+  )
+where
+
+import Control.Monad (when)
+import StrongPath
+  ( Abs,
+    Dir',
+    Path',
+    Rel,
+    (</>),
+  )
+import qualified StrongPath as SP
+import Wasp.Generator.Common (ProjectRootDir)
+import Wasp.Generator.FileDraft.Writeable
+import Wasp.Generator.FileDraft.WriteableMonad (WriteableMonad (copyDirectoryRecursive, createDirectoryIfMissing), doesDirectoryExist)
+
+-- | File draft based on another dir that is to be recursively copied.
+data CopyDirFileDraft = CopyDirFileDraft
+  { -- | Path where the dir will be copied to.
+    _dstPath :: !(Path' (Rel ProjectRootDir) Dir'),
+    -- | Absolute path of source dir to copy.
+    _srcPath :: !(Path' Abs Dir')
+  }
+  deriving (Show, Eq)
+
+instance Writeable CopyDirFileDraft where
+  write absDstDirPath draft = do
```

Comment:
```suggestion
  write projectRootAbsPath draft = do
```

because it might get confused with `dstPathAbsDir` otherwise (at least it confused me).

Reply:
Yeah that was a lift from `CopyFileDraft` and stems from use there of `dst` and `src` but agree `project` could be better (and will be helped once we rename it all for consistency at some point).